code/datums/research_upgrade_datum.dm

```diff
@@ -111,6 +112,26 @@
 	behavior = RESEARCH_UPGRADE_TIER_5 //adjust this to change amount of nanosplints in a stack, cant be higher than five, go change max_amount in the nanosplint itself, then change it.
 	upgrade_type = ITEM_ACCESSORY_UPGRADE
 
+/datum/research_upgrades/item/flamer_tank
+	name = "Upgraded Incinerator Tank"
+	desc = "An upgraded incinerator tank, with larger capacity and able to handle stronger fuels."
+	value_upgrade = 500
+	clearance_req = 1
+	change_purchase = -100
+	minimum_price = 100
+	item_reference = /obj/item/ammo_magazine/flamer_tank/custom/upgraded
+	behavior = null
+	upgrade_type = ITEM_ACCESSORY_UPGRADE
+
+/datum/research_upgrades/item/flamer_tank/smoke
+	name = "Upgraded Incinerator Smoke Tank"
+	desc = "An upgraded incinerator smoke tank with a larger capacity."
+	value_upgrade = 100 //not useful enough to be expensive
+	clearance_req = 1
+	item_reference = /obj/item/ammo_magazine/flamer_tank/smoke/upgraded
+	change_purchase = 0
```

Comment:
```suggestion
	change_purchase = -100
```

Reply:
Unresolved - you need to explain why you insist on this being different than any of the other repeatables.